src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -620,6 +623,12 @@ let run_scan_files (conf : Scan_CLI.conf) (profiler : Profiler.t)
           (String_utils.unit_str (List.length cli_output.paths.scanned) "file")
           (String_utils.unit_str (List.length cli_output.results) "finding"));
 
+    (* step 6: apply autofixes *)
+    (* this must happen posterior to reporting matches, or will report the
+       already-fixed file
+    *)
+    Autofix.apply_fixes_of_core_matches res.core.results;
```

Comment:
forgot the if conf.autofix here.
Fixed in https://github.com/semgrep/semgrep/pull/9318